core/src/main/java/com/flipkart/gjex/core/filter/GjexFilter.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) The original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.flipkart.gjex.core.filter;
+
+import com.flipkart.gjex.core.logging.Logging;
+
+/**
+ * A Filter interface for processing Request, Request-Headers, Response and Response-Headers 
+ * around gRPC and HTTP method invocation
+ *
+ * @author ajay.jalgaonkar
+ */
+public abstract class GjexFilter<Req, Res, M> implements Logging {
```

Comment:
Remove the prefix `Gjex`. Classes have aboslute names, so we don't need to prefix `Gjex` since the class path itself has `gjex`

Reply:
done